doc/deploying_gcs_on_bare_metal_machines.md

```diff
@@ -0,0 +1,196 @@
+Master nodes : 1 {kube1}<br/>
+Kube nodes: 3<br/>
+
+#### Steps to install Kubernetes:
+
+1. Install ansible on master node:<br/>
+   `$ yum install ansible`
+
+2. To ensure ansible is able to ssh into the nodes, add public key of master node{kube1} as authorized
+   key in  other kube nodes<br/>
+   `$  cat ~/.ssh/id_rsa.pub | ssh root@kube2 'cat >> ~/.ssh/authorized_keys'`
+
+3. Stop firewalld on all machines<br/>
+   `$ systemctl stop firewalld`
+
+4. We will install kubernetes via the deploy-k8s script in the GCS repository. Clone GCS repository in master
+   node(kube1):<br/>
+  `$ git clone --recurse-submodules git@github.com:gluster/gcs.git`
+
+5. To install kubernetes we need create an inventory file mentioning the kubernetes nodes.<br/>
+
+   Template for GCS inventory file can be found under deploy directory in gcs repository:<br/>
+   `https://github.com/gluster/gcs/blob/master/deploy/examples/inventory-gcs-kubespray.example`
+
+   We will be deploying GCS after deploying kubernetes so we will have to provide GCS
+   nodes in the invntory file. Also, we will mention the devices that we want to use to create volumes.
+   Since we are using only 3 nodes we will provide devices for all the 3 nodes and mention
+   all the nodes in the gcs nodes.
+    ```
+    kube1 ansible_host=<node-ip> gcs_disks='["<device1>", "<device2>"]'
+    kube2 ansible_host=<node-ip> gcs_disks='["<device1>", "<device2>"]'
+    kube3 ansible_host=<node-ip> gcs_disks='["<device1>", "<device2>"]'
+
+    ## Hosts that will run etcd for the Kubernetes cluster
+    [etcd]
+    kube1
+    kube2
+    kube3
+
+    ## Hosts that will be kubernetes master nodes
+    [kube-master]
+    kube1
+
+    ## Hosts that will be kuberenetes nodes
+    [kube-node
+    kube1
+    kube2
+    kube3
+
+    ## The full kubernetes cluster
+    [k8s-cluster:children]
+    kube-master
+    kube-node
+
+    ## Hosts that will be used for GCS.
+    ## Systems grouped here need to define 'gcs_disks' as hostvars, which are the disks that will be used by GCS to 
+    provision storage.
+    [gcs-node]
+    kube1
+    kube2
+    kube3
+    ```
+
+
+6. To deploy kubernetes we need to excecute deploy-k8s.yml file using ansible and provide inventory file created in Step 5.<br/>
+   The deploy-k8s.yml file is present under deploy directory in gcs repository. <br />
+   `$ ansible-playbook -i <path to inventory file>/<name of inventory file>  <path to`
+     `deploy-k8s.yml file>/deploy-k8s.yml`
+
+7. Check whether all the nodes are in ready state.<br />
+   `$ kubectl get nodes`<br />
+
+    The status of all the nodes should be in READY state.
+
+> Note: In case deploy kubernetes fails then please check the troubleshooting section.
+
+### Deploying GCS on kubernetes cluster:
+
+#### Steps to install GCS:
+
+1. To deploy GCS on kubernetes cluster we will execute the deploy-gcs.yml file using ansible from master node(kube 1). The deploy-gcs.yml file is present under deploy directory.<br />
```

Comment:
question, if we are deploying from outside kubernetes machine, do we need to setup password-less SSH?

Reply:
Yes, the ansible requires to ssh into the machine.